models/issue.go

```diff
@@ -808,6 +823,20 @@ func (issue *Issue) ChangeAssignee(doer *User, assigneeID int64) (err error) {
 	return nil
 }
 
+// GetReactionsByType returns issue reactions grouped by type
+func (issue *Issue) GetReactionsByType() map[string]ReactionList {
+	var reactions = make(map[string]ReactionList)
+	for _, reaction := range issue.Reactions {
+		list, ok := reactions[reaction.Type]
```

Comment:
Can simplify to `reactions[reaction.Type] = append(reactions[reaction.Type], reaction)`